### DISCUSSION TITLE
在2.0.4版本中，表建立行策略后，进行left join查询涉及到添加策略的表

### DISCUSSION THREAD

**Discussion by x1481962098** [apache/doris#34251] [A - General / Q&A]
对角色分配了指定数据库的查询权限，对某些表建立行策略后，将角色分配给用户，用该用户登录后，单表查询正常，但进行left join查询涉及到添加策略的表（策略只过滤了标记字段为统一值），就会出错，如图
![image](https://github.com/apache/doris/assets/60313214/e1fa5a17-2ee3-4d04-8df0-847a7f3bac72)

**Comment by yagagagaga** [comment]
Can you show me the table's ddl and policy's ddl ?

**Reply by x1481962098** [reply]
- dwd_sales_order
CREATE TABLE
IF
	NOT EXISTS dwd_sales_order (
		`order_saleorderline_id` VARCHAR ( 255 ) ,
		`order_saleorder_id` VARCHAR ( 255 ) ,
        ......
        `order_oldsaleorder_name` VARCHAR ( 255 ),
        ......
		`@update_time` DATETIME NOT NULL DEFAULT CURRENT_TIMESTAMP,
		`@syn_flag` VARCHAR ( 1 ) NOT NULL DEFAULT 'Y',
	    `@delete_flag` VARCHAR ( 1 ) NOT NULL DEFAULT 'Y' 
	)
	UNIQUE KEY ( order_saleorderline_id )
	DISTRIBUTED BY HASH ( order_saleorderline_id ) BUCKETS 10
	PROPERTIES ( "replication_num" = "3" );

- dwm_sales_achievement_special
CREATE TABLE
IF
	NOT EXISTS dwm_sales_achievement_special(
	`createdate` INT NOT NULL,
	`order_name` VARCHAR ( 100 ),
	......
	`@update_time` DATETIME NOT NULL DEFAULT CURRENT_TIMESTAMP,
	`@syn_flag` VARCHAR ( 1 ) NOT NULL DEFAULT 'Y',
	`@delete_flag` VARCHAR ( 1 ) NOT NULL DEFAULT 'Y'
)
UNIQUE KEY (createdate,order_name, order_type, year, quarter, month )
DISTRIBUTED BY HASH ( order_name, order_type, year, quarter, month) BUCKETS auto
PROPERTIES (
"replication_num" = "3"
)

- row policy 
CREATE ROW POLICY temp_tes_dwm_sales_achievement_special  ON db.dwm_sales_achievement_special AS RESTRICTIVE TO role temp_tes USING (`@delete_flag` = 'N');

**Reply by yagagagaga** [reply]
Hi, I create a user and grant `temp_tes ` to user. But I can execute the sql normally. Can you provide more specific reproduction steps?

**Reply by x1481962098** [reply]
谢谢，那我们再看看，是不是其它的问题
